pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -552,13 +552,12 @@ def __resolve_ckpt_dir(self, trainer):
                 trainer.logger.version
                 if isinstance(trainer.logger.version, str) else f"version_{trainer.logger.version}"
             )
-
-            version, name = trainer.training_type_plugin.broadcast((version, trainer.logger.name))
```

Comment:
Do we want to broadcast the version too ?

Reply:
the version is folded into the checkpoint path so it should be included below. for deepspeed do you have separate paths per rank? then this would be an issue